File: src/x/xLinqToDB/.Tests/.v3.3.0/_VisualBasic/VisualBasicCommon.vb
﻿Imports System
Imports System.Collections.Generic
Imports System.Linq

Imports Tests.Model

Namespace VisualBasic

  Public Module VisualBasicCommon

    Public Function ParamenterName(ByVal db As ITestDataContext) As IEnumerable(Of Parent)
      Dim id As Integer
      id = 1
      Return From p In db.Parent Where p.ParentID = id Select p
    End Function

    Public Function SearchCondition1(ByVal db As ITestDataContext) As IEnumerable(Of LinqDataTypes)
      Return _
      From t In db.Types
      Where Not t.BoolValue And (t.SmallIntValue = 5 Or t.SmallIntValue = 7 Or (t.SmallIntValue Or 2) = 10)
      Select t
    End Function

    Public Function SearchCondition2(ByVal db As NorthwindDB) As IEnumerable(Of String)
      Return _
      From cust In db.Customer
      Where cust.Orders.Count > 0 And cust.CompanyName.StartsWith("H")
      Select cust.CustomerID
    End Function

    Public Function SearchCondition3(ByVal db As NorthwindDB) As IEnumerable(Of Integer)
      '#11/14/1997#
      Dim query = From order In db.Order
                  Where order.OrderDate = New DateTime(1997, 11, 14)
                  Select order.OrderID

      Return query
    End Function

    Public Function SearchCondition4(ByVal db As NorthwindDB) As IEnumerable(Of Integer)
      Dim query = From order In db.Order
                  Where #11/14/1997# = order.OrderDate
                  Select order.OrderID

      Return query
    End Function

  End Module

End Namespace

File: src/x/xLinqToDB/.Tests/.v3.3.0/_VisualBasic/VisualBasicCommon.cs
using Tests.Model;
using System;
using System.Collections.Generic;
using System.Linq;

namespace VisualBasic {
  public static class VisualBasicCommon {

    public static IEnumerable<Parent> ParamenterName(ITestDataContext db) {
      int id;
      id = 1;
      return from p in db.Parent
             where p.ParentID == id
             select p;
    }

    public static IEnumerable<LinqDataTypes> SearchCondition1(ITestDataContext db) {
      return from t in db.Types
             where !t.BoolValue & (t.SmallIntValue == 5 | t.SmallIntValue == 7 | (t.SmallIntValue | 2) == 10)
             select t;
    }

    public static IEnumerable<string> SearchCondition2(NorthwindDB db) {
      return from cust in db.Customer
             where cust.Orders.Count > 0 & cust.CompanyName.StartsWith("H")
             select cust.CustomerID;
    }

    public static IEnumerable<int> SearchCondition3(NorthwindDB db) {
      // #11/14/1997#
      var query = from order in db.Order
                  where order.OrderDate == new DateTime(1997, 11, 14)
                  select order.OrderID;

      return query;
    }

    public static IEnumerable<int> SearchCondition4(NorthwindDB db) {
      var query = from order in db.Order
                  where new DateTime(1997, 11, 14) == order.OrderDate
                  select order.OrderID;
      return query;
    }

  }

}
